doc/specs/stdlib_linked_list.md

```diff
@@ -0,0 +1,497 @@
+---
+title: linked lists
+---
+
+# The `stdlib_linked_list` module
+
+[TOC]
+
+## Introduction
+
+The `stdlib_linked_list` module defines a class and its interface to handle linked lists that
+store any type of data. The list may contain data of the same type or of various types.
+
+
+## Types
+
+### `type(linked_list)`
+
+Linked lists are variables of the type `linked_list`. The type provides all the methods
```

Comment:
Types in `stdlib` should have the suffix `_type` in `stdlib` (see e.g., `bitset_type`, `logger_type`).

```suggestion
Linked lists are variables of the derived type `linked_list_type`. The derived type provides all the methods
```